cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -49,12 +49,26 @@ public static void abortQuietly(TransactionSystemClient txClient, Transaction tx
 
   public static void invalidateQuietly(TransactionSystemClient txClient, Transaction tx) {
     try {
-      txClient.invalidate(tx.getWritePointer());
+      if (!txClient.invalidate(tx.getWritePointer())) {
+        LOG.error("Failed to invalidate transaction {}", tx);
+      }
     } catch (Throwable t) {
       LOG.error("Exception when invalidating transaction {}", tx, t);
     }
   }
 
+  /**
+   * Wraps the given {@link Throwable} as a {@link TransactionFailureException} if it is not already an instance of
+   * {@link TransactionFailureException}.
+   */
+  public static TransactionFailureException asTransactionFailure(Throwable t) {
```

Comment:
would it make sense to add some context? like "transaction failure when committing tx" or 'when flushing changes from mapper"?